> example 

        #include <iostream>
        int main() {
            try{
                throw std::exception();        
            }catch(std::exception e){
                std::cout << e.what() << std::endl;     
            }
            return 0;
        }

std::exception is the base class for the following important exceptions:

■ bad_alloc —Thrown when a request for memory using new fails


■ bad_cast —Thrown by dynamic_cast when you try to cast a wrong type
(a type that has no inheritance relation)


■ ios_base::failure —Thrown by the functions and methods in the iostream
library


--- 

#### bad_alloc : example : size of array

          #include <iostream>
          int main() {
              try{
                  std::cout << "enter size of array" << std::endl;
                  int size;
                  std::cin >> size; 
                  int* array = new int[size];        
              }catch(std::bad_alloc e){
                  std::cout << "invalid size of array : " << e.what() << std::endl;     
              }
              return 0;
          }

          output : 
          
          enter size of array
          -1
          invalid size of array : std::bad_alloc

---


####
